Marlin/src/lcd/extui/ui_api.cpp

```diff
@@ -1140,7 +1161,7 @@ namespace ExtUI {
     return isPrinting() && (isPrintingFromMediaPaused() || print_job_timer.isPaused());
   }
 
-  bool isMediaInserted() { return TERN0(HAS_MEDIA, IS_SD_INSERTED()); }
+  bool isMediaInserted() { return TERN0(HAS_MEDIA, IS_SD_INSERTED() || card.isMounted()); }
```

Comment:
For boards that lack an SD Detect pin this will always be `true` because `IS_SD_INSERTED()` always returns `true`.

Reply:
Ive found assuming true and trying to list the contents better than assuming false, and never getting a file list no matter what as I have yet to see a UI with explicit mount / unmount buttons. I did add a refresh button on mine to force a reinit, but in cases where its assumed true it will at least be able to attempt when called for.

Reply:
Coming back around to this, as a standard thing within ExtUI, this would need a new name if the behavior is really to be as proposed above, so I want to do a more thorough analysis of what ie being implied by this change. The behavior would become this:
- No SD Detect: No change. Just assumes the media is present.
- SD Inserted: All good. No need to check that it's mounted.
- SD Removed: Could be wrong, so check whether the media is mounted.

My change was to simply assert that the media must report itself as "mounted" since this would cover all cases, and also apply to the case where there is no SD Detect, which seems very sensible.

And so, my next commit will rename this method to `isMediaMounted` just to be clear.

Reply:
That name does seem more appropriate for both the new and revised function. It's essentially the ready permission to try and read.